### PR TITLE
Adding documentation on min/max-drivers-per-task config

### PR DESCRIPTION
**Pull request by SamWheating**
These config options were added back in 205 ([commit 1](https://github.com/trinodb/trino/commit/7d087ad20ac22ded87dda583bd351fd874bd0383), [commit 2](https://github.com/trinodb/trino/commit/da9b645c86dbd530c87ad9823356b3d9e40efda2)) but never made it into the documentation. 

Limiting `task.max-drivers-per-task` made a huge improvement to the throughput of our clusters, so I'd like to add these options to the documentation so that other users can discover them more easily.

### CODE REVIEW

**Comment by SamWheating** [comment]
I'll send the CLA in shortly, I just need to wait for approval from my employer.

File: docs/src/main/sphinx/admin/properties-task.rst
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-task.rst:66]: to use too many drivers, and may

**Review by mosabua — CHANGES_REQUESTED** [review]

**Comment by SamWheating** [comment]
I emailed in the CLA yesterday, any ideas why the check isn't passing?

**Comment by mosabua** [comment]
@cla-bot check

**Comment by mosabua** [comment]
@martint can you see about the CLA please?

File: docs/src/main/sphinx/admin/properties-task.rst
Comment (m57lyra) [inline docs/src/main/sphinx/admin/properties-task.rst:113]: "given there is sufficient work to do" is vague and open to interpretation. Please clarify.

**Review by m57lyra — APPROVED** [review]

File: docs/src/main/sphinx/admin/properties-task.rst
Comment (rosewms) [inline docs/src/main/sphinx/admin/properties-task.rst:68]: This sentence sounds a little awkward and I wasn't entirely sure what the last sentence meant. What about:

> "Controls the maximum number of drivers a task runs concurrently. Setting this value reduces the likelihood that a > task uses too many drivers and can improve concurrent query performance. This can lead to resource waste if it runs too few concurrent queries."

File: docs/src/main/sphinx/admin/properties-task.rst
Comment (SamWheating) [reply docs/src/main/sphinx/admin/properties-task.rst]: Good point. I had a quick read through the source code / other documentation and have updated it to 
```
The minimum number of drivers guaranteed to run concurrently for a single task given
the task has remaining splits to process.
```
Does this sound specific enough?

File: docs/src/main/sphinx/admin/properties-task.rst
Comment (SamWheating) [reply docs/src/main/sphinx/admin/properties-task.rst:68]: I think you're right, but what if we combined the two sentences, like:

> Controls the maximum number of drivers a task runs concurrently. Setting this value reduces the likelihood that a task uses too many drivers and can improve concurrent query performance but can lead to resource waste if there are few concurrent queries.

Thoughts?

Comment (SamWheating) [reply docs/src/main/sphinx/admin/properties-task.rst:68]: Ok, that sounds good to me - I'll update with your original suggestion. I should also read up on the google dev writing style guide as it sounds like a good reference.

File: docs/src/main/sphinx/admin/properties-task.rst
Comment (electrum) [inline docs/src/main/sphinx/admin/properties-task.rst:106]: Can you fix this to be the same width as the line above?

File: docs/src/main/sphinx/admin/properties-task.rst
Comment (electrum) [inline docs/src/main/sphinx/admin/properties-task.rst:59]: Can you fix this to be the same width as the line above?

**Review by electrum — APPROVED** [review]

**Comment by electrum** [comment]
Please squash into a single commit when updating the PR.

**Comment by electrum** [comment]
Thanks!

**Pull request merged**
